### PR TITLE
Reorg config.

### PR DESCRIPTION
**Pull request by jovany-wang**
- [x] deprecated specifying config per party and also remove the related unit test.
- [x] made all face config API as dict type instead of an internal strong type.
- [x] simplified param address.

### CODE REVIEW

File: tests/test_listen_addr.py [hunk 1]
```diff
@@ -36,11 +36,11 @@ def get_value(self):
 
 def run(party, is_inner_party):
     compatible_utils.init_ray(address='local')
-    cluster = {
-        'alice': {'address': '127.0.0.1:11012', 'listen_addr': '0.0.0.0:11012'},
```
Comment (NKcqx) [inline tests/test_listen_addr.py:40]: As a result, `addresses` still can't contain just the address of party, but a dict where `listening_on` fields (and  others maybe) also supported

**Review by NKcqx — COMMENTED** [review]
Should we keep `addresses` as simple as possible, i.e. `addresses = {'alice': str, 'bob': str}`, instead of a nested dict.

**Comment by jovany-wang** [comment]
@NKcqx Then how can we make listening address work?

File: fed/proxy/grpc/grpc_proxy.py [hunk 2]
```diff
@@ -142,7 +142,7 @@ def get_grpc_config_by_party(self, dest_party):
         grpc_metadata = self._grpc_metadata
         grpc_options = self._grpc_options
 
-        dest_party_msg_config = self._cluster[dest_party].get(
+        dest_party_msg_config = self._addresses[dest_party].get(
```
Comment (fengsp) [inline fed/proxy/grpc/grpc_proxy.py:145]: So we still support this party config in `addresses`?

File: tests/test_listen_addr.py (hunk 1 above)
Comment (fengsp) [reply tests/test_listen_addr.py:40]: We could add a simple helper here and support both `str` and `dict` as addresses value, if you think `listen_addr` and `address` are the same in common cases

File: fed/api.py [hunk 3]
```diff
@@ -109,50 +105,46 @@ def init(
...
     tls_config = {} if tls_config is None else tls_config
     if tls_config:
         assert (
             'cert' in tls_config and 'key' in tls_config
         ), 'Cert or key are not in tls_config.'
 
-    global_cross_silo_message_config = \
-        global_cross_silo_message_config or CrossSiloMessageConfig()
+    cross_silo_message_dict = config.get("cross_silo_message", {})
+    cross_silo_message_config = GrpcCrossSiloMessageConfig.from_dict(
```
Comment (fengsp) [inline fed/api.py:134]: why is this a `GrpcCrossSiloMessageConfig`, since we are having `sender_proxy_cls`, I believe we also need `proxy_config_cls`?

**Comment by NKcqx** [comment]
> @NKcqx Then how can we make `listening addresses` work?

Since `listen_addr` is used inside ReceiveProxy, i.e. cross-silo communication, and it's kinda like an advanced config, what do you think we move it to `CrossSiloMessageConfig`?

File: fed/proxy/grpc/grpc_proxy.py (hunk 2 above)
Comment (NKcqx) [reply fed/proxy/grpc/grpc_proxy.py:145]: For now, yes.
But party-specific config will be deprecated soon or later.

**Comment by jovany-wang** [comment]
> Since listen_addr is used inside ReceiveProxy, i.e. cross-silo communication, and it's kinda like an advanced config, what do you think we move it to CrossSiloMessageConfig?

I don't have idea yet. Let's move forward as what you described here.

File: fed/proxy/grpc/grpc_proxy.py (hunk 2 above)
Comment (jovany-wang) [reply fed/proxy/grpc/grpc_proxy.py:145]: I deprecated it in this PR.

File: fed/api.py (hunk 3 above)
Comment (jovany-wang) [reply fed/api.py:134]: Yes. But it's not related to this PR. We should finish it at https://github.com/ray-project/rayfed/issues/159

**Comment by jovany-wang** [comment]
@fengsp @NKcqx 
Let us move listening addressed into cross silo config as a workaround right now.

**Comment by fengsp** [comment]
> @fengsp @NKcqx Let us move listening addresses into cross silo config as a workaround right now.

I believe this is right, since we do not need to config `listen_addr` for other parties at all

**Comment by jovany-wang** [comment]
@NKcqx @fengsp @zhouaihui  Please help another review.

**Review by fengsp — APPROVED** [review]
LGTM

**Pull request merged**